src/arrayMethodJoin.js

```diff
@@ -5,8 +5,40 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator === undefined && this.length >= 2) {
+      for (let i = 1; i < this.length; i++) {
+        result += `,${this[i]}`;
+      }
+    } else if (this.length >= 2) {
+      for (let i = 1; i < this.length; i++) {
+        if (this[i] === null || this[i] === undefined) {
+          result += ',';
+        } else {
+          result += `${separator}${this[i]}`;
+        }
+      }
+    }
```

Comment:
you can also do that using 1 loop - [checklist ](https://github.com/mate-academy/js_array-method-join/blob/master/checklist.md) point 5
```suggestion
   for(let i = 0; i < this.length; i++) {
       if(this[i] !== null && this[i] !== undefined) {
          // add letter to our result
       }
       
       if(i is smaller than our length) {
         // add separator to new string
       }
   }
```